app/controllers/trainees/programme_details_controller.rb

```diff
@@ -3,9 +3,15 @@
 module Trainees
   class ProgrammeDetailsController < ApplicationController
     PROGRAMME_START_DATE_CONVERSION = {
-      "programme_start_date(3i)" => "day",
-      "programme_start_date(2i)" => "month",
-      "programme_start_date(1i)" => "year",
+      "programme_start_date(3i)" => "start_day",
+      "programme_start_date(2i)" => "start_month",
+      "programme_start_date(1i)" => "start_year",
+    }.freeze
+
+    PROGRAMME_END_DATE_CONVERSION = {
```

Comment:
**Refactor**

1. You can combine this conversion hash with `PROGRAMME_START_DATE_CONVERSION`, and just have one `PROGRAMME_DATE_CONVERSION` hash.

2. Then you could have just one method `programme_date_params` instead of one method for `programme_end_date_params` and one for `def programme_start_date_params` that do the same thing.

3. Then you wouldn't need two splats and two merges in `programme_details_params`

Reply:
Thanks, done.